### PR TITLE
feat(instr-redis-4): add @opentelemetry/instrumentation-redis-4

### PR DESCRIPTION
**Pull request by david-luna**
Closes: #38

### CODE REVIEW

File: packages/opentelemetry-node/docs/supported-technologies.md [hunk 1]
```diff
@@ -30,6 +30,8 @@ This follows from the [OpenTelemetry JS supported runtimes](https://github.com/o
 | `@opentelemetry/instrumentation-winston` | Instruments `winston` package for version range `>1 <4`         | [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-winston#readme) |
 | `@opentelemetry/instrumentation-tedious` | Instruments `tedious` package for version range `>=1.11.0 <=15` | [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-tedious#readme) |
 | `@opentelemetry/instrumentation-undici`  | Instruments `undici` package for version range `>=5.12.0`       | [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/instrumentation-undici#readme) |
+| `@opentelemetry/instrumentation-aws-sdk`  | Instruments `aws-sdk` package for version range `>=2 <4`       | [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#readme) |
```
Comment (david-luna) [inline packages/opentelemetry-node/docs/supported-technologies.md:33]: note for reviewer: leftover from #37

Comment (trentm) [reply packages/opentelemetry-node/docs/supported-technologies.md:33]:
```suggestion
| `@opentelemetry/instrumentation-aws-sdk`  | Instruments `aws-sdk` version 2 and `@aws-sdk/client-*` version 3      | [README](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/plugins/node/opentelemetry-instrumentation-aws-sdk#readme) |
```

File: packages/opentelemetry-node/test/fixtures/use-redis.mjs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+// Usage: node -r @elastic/opentelemetry-node use-ioredis.mjs
```
Comment (trentm) [inline packages/opentelemetry-node/test/fixtures/use-redis.mjs:20]:
```suggestion
// Usage: node -r @elastic/opentelemetry-node use-redis.mjs
```

File: packages/opentelemetry-node/test/instr-redis-4.test.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+            t.equal(spans[4].name, 'redis-HSET');
+            t.equal(spans[5].name, 'redis-GET');
+            t.equal(spans[5].status.code, 'STATUS_CODE_ERROR');
+        },
+    },
+    // ESM redis sanity test
+    {
+        name: 'use-redis.mjs (ESM via --require)',
+        versionRanges: {
+            // TODO: issue on node docs that https://nodejs.org/api/all.html#all_module_moduleregisterspecifier-parenturl-options history doesn't show backport to v18.19.0
```
Comment (trentm) [inline packages/opentelemetry-node/test/instr-redis-4.test.js:70]:
```suggestion
```

I think we could remove this TODO comment. We already have it in the other file as a reminder that we could open a docs issue on node.js core.

**Review by trentm — APPROVED** [review]
just some nits

**Pull request merged**